### PR TITLE
Add tooltip for cell content and entries' preview in Main Table

### LINKED ISSUE #10925: Tooltip on main table cell should show full table cell content
When I hover on a table entry, there appears nothing:

image1

There should be a tool tip:

- First line: Full content of the current cell
- Second line: empty
- Third line: rendered BibTeX entry (as shown in the Preview)

BibTeX source:

```bibtex
@Article{WarnettZdun2022,
  author   = {Warnett, Stephen John and Zdun, Uwe},
  journal  = {Computer},
  title    = {Architectural {Design} {Decisions} for the {Machine} {Learning} {Workflow}},
  year     = {2022},
  issn     = {1558-0814},
  month    = mar,
  number   = {3},
  pages    = {40--51},
  volume   = {55},
  abstract = {We conducted a qualitative investigation of architectural decisions faced by practitioners and modeled current practices in machine learning. We describe a subset of the architectural decisions that were modeled, discuss uses for the model, and outline areas in which further research may be pursued.},
  doi      = {10.1109/MC.2021.3134800},
  keywords = {Computational modeling, Machine learning, Computer architecture, Quality assessment},
}
```

The second and third line should be shown only if the (new) preference is activated:

image

### PR DESCRIPTION
**Pull request by azatyamanaev**
Closes #10925

Added tooltip for Main Table cells which shows cell content and optionally entry's preview. A checkbox in Preview Tab is added to toggle on and off entry's preview.

When checkbox in Preview Tab is toggled off, only cell content is shown.

1. Toggle off "show preview in entry table tooltip" checkbox
1_show_preview_checkbox_is_toggled_off

2. Check that tooltip shows only cell content
2_tooltip_shows_only_cell_content

When checkbox in Preview Tab is toggled on, cell content and entry preview is shown

1. Toggle on "show preview in entry table tooltip" checkbox
3_show_preview_checkbox_is_toggled_on

2. Check that tooltip shows cell content and entry preview
4_tooltip_shows_cell_content_and_entry_preview

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java [hunk 1]
```diff
@@ -76,6 +78,8 @@ public MainTableColumnFactory(BibDatabaseContext database,
         this.cellFactory = new CellFactory(preferencesService, undoManager);
         this.undoManager = undoManager;
         this.stateManager = stateManager;
+        this.tooltip = new MainTableTooltip(database, dialogService, preferencesService, stateManager,
+                JabRefGUI.getThemeManager(), taskExecutor);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java:82]: Pleas pass the ThemeManager via constructor (Dependency Inversion Principle)

Comment (azatyamanaev) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java:82]: This entails adding ThemeManager as a constructor parameter for dozen other classes. Should I still do it?

Comment (azatyamanaev) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java:82]: @Siedlerchr This entails adding ThemeManager as a constructor parameter for dozen other classes. Should I still do it?

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java:82]: Yes, I know this sounds crazy but we don't have really any good dependency injection. 
But most of the calling classes should already have the theme manager, so I hope it's only a handful of classes that need ajustment

Comment (azatyamanaev) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java]: @Siedlerchr I think it's 17 files
[commit](https://github.com/JabRef/jabref/pull/11176/commits/04774ccaf8fbd7e1d7630e088aebc8069d0d51ac)

**Comment by Siedlerchr** [comment]
lgtm
Personally I find this annoying but @koppor  wished for this ;)

**Review by Siedlerchr — DISMISSED** [review]

File: src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java (hunk 1 above)
Comment (calixtus) [reply src/main/java/org/jabref/gui/maintable/MainTableColumnFactory.java]:
```
ThemeManager themeManager = Injector.instantiateModelOrService(ThemeManager.class);
```
is a temporary solution until we have a proper di solution.

**Review by calixtus — CHANGES_REQUESTED** [review]
Please revert the mass injection and use the temporary di solution.

**Review by Siedlerchr — APPROVED** [review]

**Review by calixtus — APPROVED** [review]

**Pull request merged**